ui/apps/akasha/src/components/feed-page/feed-page.tsx

```diff
@@ -41,6 +41,20 @@ const FeedPage: React.FC<FeedPageProps & RootComponentProps> = props => {
 
   const [analyticsActions] = useAnalytics();
 
+  //get the post id for repost from the search param
+  const [postId, setPostId] = React.useState(new URLSearchParams(location.search).get('repost'));
+
+  React.useEffect(() => {
+    /*The popstate listener is required for reposts happening from the feed page */
+    const popStateHandler = () => {
+      setPostId(new URLSearchParams(location.search).get('repost'));
+    };
+    window.addEventListener('popstate', popStateHandler);
+    return () => {
+      window.removeEventListener('popstate', popStateHandler);
```

Comment:
Add an AbortController signal to the addEventListener and here you can just call `signal.abort()` to remove the listener.

Reply:
Done